scripts/globals/hunts.lua

```diff
@@ -0,0 +1,1615 @@
+  local huntId = player:getCharVar("[hunt]id")
+  local huntStatus = player:getCharVar("[hunt]status")
+  local scyldBits = bit.lshift(player:getCurrency("scyld"), 14)
+  local lockBit = bit.lshift(1, 24)
+  local registryZone = zone[player:getZoneID()]
+
+  -- one vana'diel day lockout timer after completing a hunt
+  if os.time() < player:getCharVar("[hunt]nextHunt") then
+    local bitCombo = scyldBits + lockBit
+    player:startEvent(1500,bitCombo,registryZone.huntMenu[1])
+
+  -- [hunt]status 1 player has already accepted a hunt
+  elseif huntStatus == 1 then
+    local playerHunt = zone[huntZone].hunt[huntId]
+    local pageBits = bit.lshift(playerHunt.pageId, 4)
+    local bitCombo = scyldBits + pageBits + 0x0002 -- bit displays hunt active menu
+    player:startEvent(1500,bitCombo,registryZone.huntMenu[1])
+
+  -- [hunt]status 2 player has completed a hunt
+  elseif huntStatus == 2 then
+    local playerHunt = zone[huntZone].hunt[huntId]
+    local pageBits = bit.lshift(playerHunt.pageId, 4)
+    local bitCombo = scyldBits + pageBits + 0x000A -- bit displays completion menu
+    player:startEvent(1500,bitCombo,registryZone.huntMenu[1])
+
+  -- stops player from taking a hunt if a regime is active
+  elseif player:getCharVar("[regime]id") >= 1 then
+    local bitCombo = scyldBits + 0x0001 -- bit displays regime active menu
+    player:startEvent(1500,bitCombo,registryZone.huntMenu[1])
+
+  -- default menu
+  else
+    player:startEvent(1500,scyldBits,registryZone.huntMenu[1])
+  end
+end
+
+function tpz.hunts.onEventUpdate(player,csid,option)
+  local registryZone = zone[player:getZoneID()]
+  local region = registryZone[option]
+
+  player:updateEvent(0,0,registryZone.huntMenu[option])
+
+  -- handles region select
+  option = bit.band(option, 0xFF)
+  if registryZone.huntMenu[option] then
+    player:updateEvent(0,1,registryZone.huntMenu[option])
+  end
+
+  -- gets progress of current hunt (param controls the display of kills needed)
+  if option == 3 then
+    player:updateEvent(1)
+  end
+
+  -- displays hunt info (kills required,0,0,0,zoneId,huntId,scyld bounty+fee,?)
+  if region then
+    local huntPage = registryZone.hunt[region.optionId]
+    local bountyBit = bit.lshift(huntPage.bounty, 10)
+    local feeBit = huntPage.fee
+    local scyldParam = bountyBit + feeBit
+    player:updateEvent(1,0,0,0,0,region.huntId,scyldParam,1)
+  end
+end
+
+tpz.hunts.clearHuntVars = function(player)
+  player:setCharVar("[hunt]id", 0)
+  player:setCharVar("[hunt]status", 0)
+  player:setCharVar("[hunt]zone", 0)
+  player:setCharVar("[hunt]page", 0)
+end
+
+function tpz.hunts.onEventFinish(player, csid, option)
+  local scylds = player:getCurrency("scyld")
+  local registryZone = zone[player:getZoneID()]
+  local huntEntry = registryZone.hunt[option]
+  local msg = registryZone.msg
+
+-- accepting hunt
+if huntEntry then
+    player:messageSpecial(msg[1])
+    player:messageSpecial(msg[2], huntEntry.fee, scylds)
```

Comment:
These should be moved to after the fee is applied, and updated with `player:getCurrency("scyld")`